### PR TITLE
Added PolicySettings and DeleteModal components for PolicyDetails UI

### PR DESCRIPTION
**Pull request by lobdelle**
Signed-off-by: Eric Lobdell <lobdelle@amazon.com>

### Description

Adds PolicySettings and DeleteModal components used by the PolicyDetails UI. PolicySettings is the upper render box that shows basic details of the given policy, excluding States display. DeleteModal is responsible for confirming that the user wants to delete the policy.

Screen Shot 2021-08-11 at 5 54 37 PM

Screen Shot 2021-08-11 at 5 23 17 PM

### Known Issues
- interfaces.ts file includes too many changes and may need to be reduced to just relevant changes. 

### Check List

- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: models/interfaces.ts [hunk 1]
```diff
@@ -92,8 +92,40 @@ export interface DocumentTransform {
...
+  };
+}
+
+export interface MessageTemplate {
+  source: string;
+}
+
+export interface ISMTemplate {
+  index_patterns: string[];
+  priority: number;
```
Comment (dbbaughe) [inline models/interfaces.ts:128]: Delete these and use the ones below

File: public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    super(props);
+
+    this.state = {
+      pageIndex: 0,
+      pageSize: 10,
+      showPerPageOptions: true,
+    }
+  }
+
+  // TODO: Needs to be updated to handle template array
```
Comment (dbbaughe) [inline public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx:46]: Is this TODO still true? I see it's handling ISMTemplates as an array

File: public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      })
+    })
+    return templateArray;
+  }
+
+  onTableChange = ({ page = {} }) => {
+    const { index: pageIndex, size: pageSize } = page;
+
+    this.setState({pageIndex: pageIndex});
+    this.setState({pageSize: pageSize});
```
Comment (dbbaughe) [inline public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx:62]: Combine -> `this.setState({ pagineIndex, pageSize });`

File: public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx (hunk 2 above)
Comment (lobdelle) [reply public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx:46]: Nope, removing TODO.

File: public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx [hunk 4]
```diff
@@ -43,23 +43,18 @@ export default class PolicySettings extends Component<PolicySettingsProps, Polic
...
-  // TODO: Needs to be updated to handle template array
   getTemplates = (ismTemplates: ISMTemplate[]): ISMTemplate[] => {
-    const templateArray = [];
-    ismTemplates.map((ismTemplate)=> {
-      templateArray.push({
-        indexPatterns: ismTemplate.index_patterns,
-        priority: ismTemplate.priority,
-      })
-    })
+    const templateArray = ismTemplates.map(ismTemplate => ({
```
Comment (dbbaughe) [inline public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx:47]: Previously this was renaming it to indexPatterns, but if it's not then what is the purpose of this function? Seems to take in a list of ISMTemplates and returns a list of ISMTemplates in the same format.

Comment (lobdelle) [reply public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx:47]: Good call. Removing this and simplifying.

File: public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                  <dd>{item.value}</dd>
+                </EuiText>
+              </EuiFlexItem>
+            ))}
+          </EuiFlexGrid>
+          <EuiSpacer size="s" />
+          <ContentPanel
+            panelStyles={{ padding: "20px 20px" }}
+            bodyStyles={{ padding: "10px" }}
+            title="ISM Templates"
```
Comment (bowenlan-amzn) [inline public/pages/PolicyDetails/components/PolicySettings/PolicySettings.tsx:142]: We would want `ISM Templates (number of total templates)` here

**Comment by dbbaughe** [comment]
Seems like you have to update snapshot

```
2021-08-12T01:10:41.8589340Z FAIL public/pages/PolicyDetails/components/PolicySettings/PolicySettings.test.tsx
2021-08-12T01:10:41.8591242Z   ● <PolicySettings /> spec › renders the component
2021-08-12T01:10:41.8591617Z 
2021-08-12T01:10:41.8592212Z     expect(received).toMatchSnapshot()
```

**Review by dbbaughe — APPROVED** [review]

**Pull request merged**